test/infrastructure/inmemory/internal/server/etcd/handler.go

```diff
@@ -113,8 +113,46 @@ func (m *maintenanceServer) Snapshot(_ *pb.SnapshotRequest, _ pb.Maintenance_Sna
 	return fmt.Errorf("not implemented: Snapshot")
 }
 
-func (m *maintenanceServer) MoveLeader(_ context.Context, _ *pb.MoveLeaderRequest) (*pb.MoveLeaderResponse, error) {
-	return nil, fmt.Errorf("not implemented: MoveLeader")
+func (m *maintenanceServer) MoveLeader(ctx context.Context, req *pb.MoveLeaderRequest) (*pb.MoveLeaderResponse, error) {
+	out := new(pb.MoveLeaderResponse)
+	resourceGroup, _, err := m.getResourceGroupAndMember(ctx)
+	if err != nil {
+		return nil, err
+	}
+	etcdPods := &corev1.PodList{}
+	cloudClient := m.manager.GetResourceGroup(resourceGroup).GetClient()
+	if err := cloudClient.List(ctx, etcdPods,
+		client.InNamespace(metav1.NamespaceSystem),
+		client.MatchingLabels{
+			"component": "etcd",
+			"tier":      "control-plane"},
+	); err != nil {
+		return nil, errors.Wrap(err, "failed to list etcd members")
+	}
+
+	if len(etcdPods.Items) == 0 {
+		return nil, nil
+	}
+
+	for i := range etcdPods.Items {
+		pod := &etcdPods.Items[i]
+		for k, v := range pod.GetAnnotations() {
+			if k == cloudv1.EtcdMemberIDAnnotationName {
+				target := strconv.FormatInt(int64(req.TargetID), 10)
+				if v == target {
+					updatedPod := pod.DeepCopy()
+					annotations := updatedPod.GetAnnotations()
+					annotations[cloudv1.EtcdLeaderFromAnnotationName] = time.Now().Format(time.RFC3339)
+					updatedPod.SetAnnotations(annotations)
+					err := cloudClient.Patch(ctx, updatedPod, client.MergeFrom(pod))
+					if err != nil {
+						return nil, err
+					}
+				}
```

Comment:
Let's return success
```suggestion
				     return out, nil
				}
```